adafruit_tinylora/adafruit_tinylora.py

```diff
@@ -0,0 +1,405 @@
+        self.region = country
+
+    @property
+    def country(self):
+        """Returns the TTN Frequency Country.
+        """
+        return self.region
+
+    @property
+    def device_address(self):
+        """Returns the TTN Device Address.
+        """
+        return self.dev_addr
+
+    @property
+    def application_key(self):
+        """Returns the TTN Application Key.
+        """
+        return self.app_key
+
+    @property
+    def network_key(self):
+        """Returns the TTN Network Key.
+        """
+        return self.net_key
+
+
+# pylint: disable=too-many-instance-attributes
+class TinyLoRa:
+    """TinyLoRa Interface
+    """
+    # SPI Write Buffer
+    _BUFFER = bytearray(2)
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, spi, cs, irq, ttn_config, channel=None):
+        """Interface for a HopeRF RFM95/6/7/8(w) radio module. Sets module up for sending to
+        The Things Network.
+
+        :param ~busio.SPI spi: The SPI bus the device is on
+        :param ~digitalio.DigitalInOut cs: Chip select pin (RFM_NSS)
+        :param ~digitalio.DigitalInOut irq: RFM's DIO0 Pin (RFM_DIO0)
+        :param TTN ttn_config: TTN Configuration.
+        :param int channel: Frequency Channel.
+        """
+        self._irq = irq
+        # Set up SPI Device on Mode 0
+        self._device = spi_device.SPIDevice(spi, cs, baudrate=4000000,
+                                            polarity=0, phase=0)
+        # Set Frequency registers
+        self._rfm_msb = None
+        self._rfm_mid = None
+        self._rfm_lsb = None
+        # Set datarate registers
+        self._sf = None
+        self._bw = None
+        self._modemcfg = None
+        self.set_datarate("SF7BW125")
+        # Set regional frequency plan
+        if 'US' in ttn_config.country:
+            from adafruit_tinylora.ttn_usa import TTN_FREQS
+            self._frequencies = TTN_FREQS
+        elif adafruit_tinylora.ttn_config.country == 'AS':
+            from adafruit_tinylora.ttn_as import TTN_FREQS
+            self._frequencies = TTN_FREQS
+        elif adafruit_tinylora.ttn_config.country == 'AU':
+            from adafruit_tinylora.ttn_au import TTN_FREQS
+            self._frequencies = TTN_FREQS
+        elif adafruit_tinylora.ttn_config.country == 'EU':
+            from adafruit_tinylora.ttn_eu import TTN_FREQS
+            self._frequencies = TTN_FREQS
+        else:
+            print("Country Code Incorrect/Unsupported")
+        # Set Channel Number
+        self._channel = channel
+        self._tx_random = randint(0, 7)
+        if self._channel is not None:
+            # set single channel
+            self.set_channel(self._channel)
+        # Init FrameCounter
+        self.frame_counter = 0
+        # Verify the version of the RFM module
+        self._version = self._read_u8(0x42)
+        if self._version != 18:
+            print("Error Detecting RFM95W.")
+            print(" Check your wiring.")
```

Comment:
Move this check to right after you create your `SPIDevice` above. Also, throw an exception since this indicates something is wrong and you can't really continue.

Reply:
Good idea, raising a runtimeerror after spidevice creation instead.